docs/gpu-acceleration.md

```diff
@@ -126,7 +126,7 @@ export PATH=$PATH:$CUDA_HOME/bin
 
 # Recompile llama-cpp-python using CUDA
 pip cache remove llama_cpp_python
-pip install --force-reinstall llama_cpp_python==0.2.79 -C cmake.args="-DLLAMA_CUDA=on"
+pip install --force-reinstall "numpy<2.0" llama_cpp_python==0.2.79 -C cmake.args="-DLLAMA_CUDA=on"
```

Comment:
Does `--no-deps` work for you, too?
```suggestion
pip install --force-reinstall --no-deps llama_cpp_python==0.2.79 -C cmake.args="-DLLAMA_CUDA=on"
```

Reply:
Hi @tiran, thanks for the reply. I'm trying to reproduce the issue but I'm unable to. I guess I'll have to set up a new Arch instance. I'll let you know how it goes.

Reply:
Hi @tiran, I just had the same issue on Fedora 40 when running the command below:

```CUDAHOSTCXX=$(which clang++-17) pip install --force-reinstall llama_cpp_python==0.2.79 -C cmake.args="-DLLAMA_CUDA=on"```

Output:
```
Collecting llama_cpp_python==0.2.79
  Using cached llama_cpp_python-0.2.79.tar.gz (50.3 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting typing-extensions>=4.5.0 (from llama_cpp_python==0.2.79)
  Using cached typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting numpy>=1.20.0 (from llama_cpp_python==0.2.79)
  Using cached numpy-2.0.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)
Collecting diskcache>=5.6.1 (from llama_cpp_python==0.2.79)
  Using cached diskcache-5.6.3-py3-none-any.whl.metadata (20 kB)
Collecting jinja2>=2.11.3 (from llama_cpp_python==0.2.79)
  Using cached jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting MarkupSafe>=2.0 (from jinja2>=2.11.3->llama_cpp_python==0.2.79)
  Using cached MarkupSafe-2.1.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.0 kB)
Using cached diskcache-5.6.3-py3-none-any.whl (45 kB)
Using cached jinja2-3.1.4-py3-none-any.whl (133 kB)
Using cached numpy-2.0.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (19.3 MB)
Using cached typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Using cached MarkupSafe-2.1.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (28 kB)
Building wheels for collected packages: llama_cpp_python
  Building wheel for llama_cpp_python (pyproject.toml) ... done
  Created wheel for llama_cpp_python: filename=llama_cpp_python-0.2.79-cp311-cp311-linux_x86_64.whl size=177747534 sha256=b09fc2a76dd6871277a303f952bf3cd467d3d256702f2fba69df84df101882b8
  Stored in directory: /home/mdevino/.cache/pip/wheels/a9/d4/e0/9f5a5f0fc639dffa4cb52eb13132b4aa3bc5bf1ab6bae61bfd
Successfully built llama_cpp_python
Installing collected packages: typing-extensions, numpy, MarkupSafe, diskcache, jinja2, llama_cpp_python
  Attempting uninstall: typing-extensions
    Found existing installation: typing_extensions 4.12.2
    Uninstalling typing_extensions-4.12.2:
      Successfully uninstalled typing_extensions-4.12.2
  Attempting uninstall: numpy
    Found existing installation: numpy 1.26.4
    Uninstalling numpy-1.26.4:
      Successfully uninstalled numpy-1.26.4
  Attempting uninstall: MarkupSafe
    Found existing installation: MarkupSafe 2.1.5
    Uninstalling MarkupSafe-2.1.5:
      Successfully uninstalled MarkupSafe-2.1.5
  Attempting uninstall: diskcache
    Found existing installation: diskcache 5.6.3
    Uninstalling diskcache-5.6.3:
      Successfully uninstalled diskcache-5.6.3
  Attempting uninstall: jinja2
    Found existing installation: Jinja2 3.1.4
    Uninstalling Jinja2-3.1.4:
      Successfully uninstalled Jinja2-3.1.4
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
transformers 4.42.4 requires numpy<2.0,>=1.17, but you have numpy 2.0.0 which is incompatible.
accelerate 0.32.1 requires numpy<2.0.0,>=1.17, but you have numpy 2.0.0 which is incompatible.
trl 0.9.6 requires numpy<2.0.0,>=1.18.2, but you have numpy 2.0.0 which is incompatible.
instructlab-training 0.2.0 requires numpy<2.0.0,>=1.26.4; python_version != "3.10", but you have numpy 2.0.0 which is incompatible.
instructlab 0.17.1.dev503 requires numpy<2.0.0,>=1.26.4; python_version >= "3.11", but you have numpy 2.0.0 which is incompatible.
Successfully installed MarkupSafe-2.1.5 diskcache-5.6.3 jinja2-3.1.4 llama_cpp_python-0.2.79 numpy-2.0.0 typing-extensions-4.12.2
```

Using `--no-deps` seem to have worked (after downgrading numpy to 1.26.4 again):

```
CUDAHOSTCXX=$(which clang++-17) pip install --force-reinstall --no-deps llama_cpp_python==0.2.79 -C cmake.args="-DLLAMA_CUDA=on"
```

Output:

```
Collecting llama_cpp_python==0.2.79
  Using cached llama_cpp_python-0.2.79-cp311-cp311-linux_x86_64.whl
Installing collected packages: llama_cpp_python
  Attempting uninstall: llama_cpp_python
    Found existing installation: llama_cpp_python 0.2.79
    Uninstalling llama_cpp_python-0.2.79:
      Successfully uninstalled llama_cpp_python-0.2.79
Successfully installed llama_cpp_python-0.2.79
```

I'm updating the change based on your suggestion.